hera_cal/datacontainer.py

```diff
@@ -181,40 +182,128 @@ def concatenate(self, D, axis=0):
         return DataContainer(newD)
 
     def __add__(self, D):
```

Comment:
I don't think that's necessary to be honest: this is an operator overload (hence the `__`) so the user shouldn't be using these functions directly. This overloads the `+` logic so that I can add two datacontainers as if they were regular python objects.

Reply:
I understand that you're overloading the `+` operator, but you _should_ still explicitly tell me (i.e., the user) what the code is expecting `D` to be. Because you're not doing explicit type-checking (which can be fine, but doesn't give me any hints about its type), you should at least document what is supported. For instance, in another comment, you said it _could_ be scalar, but that wasn't obvious to me from reading the code.

Reply:
sure thing